azure/converters/image_test.go

```diff
@@ -31,6 +31,25 @@ func Test_ImageToPlan(t *testing.T) {
 		image  *infrav1.Image
 		expect func(*GomegaWithT, *compute.Plan)
 	}{
+		{
+			name: "Should return a plan for a Community Gallery image with plan details",
+			image: &infrav1.Image{
+				ComputeGallery: &infrav1.AzureComputeGalleryImage{
+					Plan: &infrav1.ImagePlan{
+						Publisher: "my-publisher",
+						Offer:     "my-offer",
+						SKU:       "my-sku",
+					},
+				},
+			},
+			expect: func(g *GomegaWithT, result *compute.Plan) {
+				g.Expect(result).To(Equal(&compute.Plan{
+					Name:      to.StringPtr("my-sku"),
+					Publisher: to.StringPtr("my-publisher"),
+					Product:   to.StringPtr("my-offer"),
```

Comment:
not sure if this is correct, productname and offer name are not the same.
https://docs.microsoft.com/en-us/rest/api/compute/virtual-machines/create-or-update#plan

Reply:
This is just a copy-paste from other test, I think this is the local convention, even if it's not a 1:1 mapping with Azure API.

Reply:
@agrawalkshitij https://docs.microsoft.com/en-us/rest/api/compute/virtual-machines/create-or-update#plan states explicitly states "This is the same value as Offer under the imageReference element", are we misunderstanding?

Reply:
Is Name also supposed to be Sku name? I don't think there is a 1:1 mapping like this from pub/off/sku to publisher/product/Name .. I can double check with my team.
I do see the doc mentions about product == offer, but not the name.

Reply:
since this test code, i guess it's fine.